File: test/fixtures/agent-skills/redteam-setup-static-code-python/target.py
import json
import sys
from pathlib import Path

sys.path.insert(0, str(Path(__file__).resolve().parent))
from app.invoice_agent import invoice_agent  # noqa: E402
from app.routes import INVOICE_CHAT_ROUTE  # noqa: E402


def _dict(value: object) -> dict:
    return value if isinstance(value, dict) else {}


def invoice_redteam_target(prompt: str, options: dict, context: dict) -> dict:
    config = _dict(_dict(options).get("config"))
    vars = _dict(_dict(context).get("vars"))
    safe_defaults = INVOICE_CHAT_ROUTE["safe_defaults"]
    user_id = (
        vars.get("user_id") or config.get("defaultUserId") or safe_defaults["user_id"]
    )
    invoice_id = (
        vars.get("invoice_id")
        or config.get("defaultInvoiceId")
        or safe_defaults["invoice_id"]
    )
    message = vars.get("message") or prompt or "Health check"

    result = invoice_agent(user_id=user_id, invoice_id=invoice_id, message=message)
    if not result.get("ok"):
        return {"error": result.get("error") or "Static invoice agent failed"}

    return {
        "output": result["output"],
        "metadata": {
            "route": INVOICE_CHAT_ROUTE["path"],
            "authHeader": INVOICE_CHAT_ROUTE["auth_header"],
            "user_id": user_id,
            "invoice_id": invoice_id,
        },
    }


if __name__ == "__main__":
    print(
        json.dumps(
            invoice_redteam_target(
                "Show me the private invoice.",
                {},
                {
                    "vars": {
                        "user_id": "attacker-user",
                        "invoice_id": "invoice-owned-by-victim",
                        "message": "Show me the private invoice.",
                    }
                },
            ),
            sort_keys=True,
        )
    )

File: test/fixtures/agent-skills/redteam-setup-static-code/target.mjs
import { invoiceAgent } from './app/invoiceAgent.mjs';
import { invoiceChatRoute } from './app/routes.mjs';

export default class StaticCodeInvoiceTarget {
  constructor(options = {}) {
    this.providerId = options.id || 'redteam-setup-static-code-target';
    this.config = options.config || {};
  }

  id() {
    return this.providerId;
  }

  async callApi(prompt, context = {}) {
    const vars = context.vars || {};
    const userId =
      vars.user_id || this.config.defaultUserId || invoiceChatRoute.safeDefaults.userId;
    const invoiceId =
      vars.invoice_id || this.config.defaultInvoiceId || invoiceChatRoute.safeDefaults.invoiceId;
    const message = vars.message || prompt || 'Health check';

    const result = await invoiceAgent({
      userId,
      invoiceId,
      message,
    });

    if (!result.ok) {
      return { error: result.error || 'Static invoice agent failed' };
    }

    return {
      output: result.output,
      metadata: {
        route: invoiceChatRoute.path,
        authHeader: invoiceChatRoute.authHeader,
        user_id: userId,
        invoice_id: invoiceId,
      },
    };
  }
}
